mmrotate/core/mask/structures.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import cv2
+import numpy as np
+from mmdet.core.mask import BitmapMasks
+
+
+class RBitmapMasks(BitmapMasks):
+    """This class represents masks in the form of bitmaps. Compared to the
+    original class, this class supports getting the minimum area rectangles
+    from masks.
+
+    Args:
+        masks (ndarray): ndarray of masks in shape (N, H, W), where N is
+            the number of objects.
+        height (int): height of masks
+        width (int): width of masks
+    """
+
+    def get_rbboxes(self):
+        num_masks = len(self)
+        rboxes = np.zeros((num_masks, 5), dtype=np.float32)
+        x_any = self.masks.any(axis=1)
+        y_any = self.masks.any(axis=2)
+        for idx in range(num_masks):
+            x = np.where(x_any[idx, :])[0]
+            y = np.where(y_any[idx, :])[0]
+            if len(x) > 0 and len(y) > 0:
+                contours = cv2.findContours(self.masks[idx], cv2.RETR_TREE,
+                                            cv2.CHAIN_APPROX_SIMPLE)[0][0]
+                (cx, cy), (w, h), a = cv2.minAreaRect(contours)
+                rboxes[idx, :] = np.array(
+                    [cx, cy, w, h, np.radians(a)], dtype=np.float32)
```

Comment:
Here, we need to support `mmrotate.core.bbox.poly2obb` for various angle versions.
However, `poly2obb` supposes that polygons have 4 vertices.
How can we support this?

Reply:
In mmrotate1.x, we supposes all rotated boxes use `oc` angle version during data transforms. 
We only need to distinguish different angle versions within the box_head.
So I think it's OK to write like this.